### PR TITLE
feat: added a basic "UI" and functions for interaction with the console

### PR DESCRIPTION
**Pull request by NilEis**
### Additions

- Functions for setting the color of the console
grafik
- Functions for clearing the console
- A basic main menu

### CODE REVIEW

File: include/RSA.h [hunk 1]
```diff
@@ -7,14 +7,14 @@
...
- * \param ret_d is a pointer to the uint32_t of the d value of the private key
- *
+/**
+ * @brief Generates RSA key pairs from two prime numbers
+ * @param p the first prime number
+ * @param q the secound prime number
+ * @param ret_N the pointer to the uint32_t of the n value of the public and private key
+ * @param ret_e the pointer to the uint32_t of the e value of the public key
+ * @param ret_d the pointer to the uint32_t of the d value of the private key
+ * @return (void)
```
Comment (NilEis) [inline include/RSA.h:17]: doxygen-style

File: include/console.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+/**
+ * @brief resets the color of the console to black and white
+ * @return (void)
+ */
+void console_reset_color(void);
+
+#if defined(_WIN32) || defined(_WIN64)
+
+#define c_printf printf
```
Comment (NilEis) [inline include/console.h:29]: Colors are added by changing the console and not by using Ansii colour codes

File: include/console.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+#elif defined(__unix__)
+//replace c_printf with the colored text
+#define c_printf(format, args...)                       \
+    do                                                  \
+    {                                                   \
+        char *str = (char *)malloc(256 * sizeof(char)); \
+        sprintf(str, format, ##args);                   \
+        printf("%s%s\x1b[0m", s_color, str);            \
+        free(str);                                      \
+    } while (0);
```
Comment (NilEis) [inline include/console.h:60]: adds the Ansii color codes to the string

File: src/console.c [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+/**
+ * @brief Resets the colors to black and white
+ * @return (void)
+ */
+void console_reset_color(void)
+{
+    sprintf(s_color, "\x1b[%d;%dm", 0, 37);
+}
```
Comment (NilEis) [inline src/console.c:35]: Should work?!

File: src/menu.c [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+            break;
+        case 0x0D: //RETURN
+        case 0x20: //SPACE
+            key = 5;
+            break;
+        case 0x1B: //ESCAPE
+            key = 6;
+            break;
+        }
+#endif
```
Comment (NilEis) [inline src/menu.c:148]: No idea how to do this without getch()

File: src/menu.c [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+            {
+                key = 6; //ESCAPE
+            }
+            break;
+        case 0x0A: //RETURN
+        case 0x20: //SPACE
+            key = 5;
+            break;
+        }
+        tcsetattr(0, TCSANOW, &term_old);
```
Comment (NilEis) [inline src/menu.c:136]: Works in WSL

**Review by jonsch318 — APPROVED** [review]

**Pull request merged**